backend/src/main/java/com/bakdata/conquery/models/forms/managed/RelativeFormQueryPlan.java

```diff
@@ -66,22 +67,37 @@ public EntityResult execute(QueryExecutionContext ctx, Entity entity) {
 		// We look at the first result line to determine the length of the subresult
 		int featureLength = featureResult.getValues().get(0).length;
 		int outcomeLength = outcomeResult.getValues().get(0).length;
-		// Whole result is the concatenation of the subresults. However the sub result includes the date restriction which we drop.
-		int size = featureLength + outcomeLength - 2;
-
-		// merge the full (index == null) lines
-		Object[] mergedFull = new Object[size];
-		setFeatureValues(mergedFull, featureResult.getValues().get(0));
-		setOutcomeValues(mergedFull, outcomeResult.getValues().get(0), featureLength);
-		values.add(mergedFull);
+		
+		/*
+		 *  Whole result is the concatenation of the subresults. The final output format combines resolution info, index and eventdate of both sub queries.
+		 *  The feature/outcome sub queries are of in form of: [RESOLUTION], [INDEX], [EVENTDATE], [FEATURE/OUTCOME_DR], [FEATURE/OUTCOME_SELECTS]... 
+		 *  The wanted format is: [RESOLUTION], [INDEX], [EVENTDATE], [FEATURE_DR], [OUTCOME_DR], [FEATURE_SELECTS]... , [OUTCOME_SELECTS]
+		 */
+		int size = featureLength + outcomeLength - 3/*= [RESOLUTION], [INDEX], [EVENTDATE]*/;
+
+		int resultStartIndex = 0;
+		if(contexts.size()>=2
+			&& contexts.get(0).getSubdivisionMode().equals(DateContextMode.COMPLETE)
```

Comment:
With a complex if condition like this, it's always useful to extract the entire condition into a variable (or function) to give it a name that explains it

Reply:
Danke Kai, pass ich an